security/login_link.rst

```diff
@@ -0,0 +1,510 @@
+        $container->loadFromExtension('security', [
+            'firewalls' => [
+                'main' => [
+                    'login_link' => [
+                        'check_route' => 'login_check',
+                        // only allow the link to be used once
+                        'max_uses' => 1,
+
+                        // optionally, configure the cache pool
+                        //'used_link_cache' => 'cache.redis',
+                    ],
+                ],
+            ],
+        ]);
+
+.. caution::
+
+    When setting ``max_uses`` to ``1``, you must take extra precautions to
+    make it work as expected. Email providers and browsers often load a
+    preview of the links, meaning that the link is already invalidated by
+    the preview loader.
+
+    In order to solve this issue, add a page between the link the user
+    clicks and the ``check_route`` page. On this page, add a button or
+    action the user can use to login::
+
+        // src/Controller/SecurityController.php
+        namespace App\Controller;
+
+        // ...
+        use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+        class SecurityController extends AbstractController
+        {
+            // ...
+
+            /**
+             * @Route("/process_login", name="process_login")
+             */
+            public function processLink(Request $request, UrlGeneratorInterface $urlGenerator)
+            {
+                // get the login link query parameters and generate the `check_route` URL
+                $expires = $request->query->get('expires');
+                $username = $request->query->get('user');
+                $hash = $request->query->get('hash');
+
+                $checkUrl = $urlGenerator->generate('login_check', [
+                    'expires' => $expires,
+                    'user' => $username,
+                    'hash' => $hash,
+                ]);
+
+                // render a template with a link to the check route
+                return $this->render('security/process_login.html.twig', [
+                    'check_url' => $checkUrl,
+                ]);
+            }
+
+            /**
+             * @Route("/login", name="login")
+             */
+            public function requestLoginLink(LoginLinkHandlerInterface $loginLinkHandler, UserRepository $userRepository, Request $request)
+            {
+                if ($request->isMethod('POST')) {
+                    $email = $request->request->get('email');
+                    $user = $userRepository->findOneBy(['email' => $email]);
+
+                    // create a login link for $user
+                    $loginLinkDetails = $loginLinkHandler->createLoginLink($user);
+
+                    // create a link to the process link page instead
+                    $queryParameters = parse_url($loginLinkDetails->getUrl(), \PHP_URL_QUERY);
+                    $loginLink = $urlGenerator->generate('process_login').'?'.$queryParameters;
```

Comment:
This code is quite a bit more complex than I would like. Should we add a `LoginLinkDetails::getParameters()` method to make it more easy to add a page in-between? @weaverryan

Reply:
Another option to make this completely solution less complex would be to add a `require_method` or the like option. If you set this to `require_method: POST`, the authenticator will only catch the `check_route` when it's called with POST. That allows you to implement the in-between page in the `checkLogin` controller directly.

Reply:
I really, really like that `require_method: POST` idea. That would get my vote!